crates/provider-sqldb-postgres/wit/deps/postgres/query.wit

```diff
@@ -0,0 +1,48 @@
+package wasmcloud:postgres@0.1.0-draft;
+
+/// Interface for querying a Postgres database
+interface query {
+  use types.{pg-value, result-row, query-error};
+
+  /// Query a Postgres database, leaving connection/session management
+  /// to the callee/implementer of this interface (normally a provider configured with conneciton credentials)
+  ///
+  /// Queries *must* be parameterized, with named arguments in the form of `$<integer>`, for example:
+  ///
+  /// ```
+  /// SELECT email,username FROM users WHERE uuid=$1;
+  /// ```
+  ///
+  query: func(query: string, params: list<pg-value>) -> result<list<result-row>, query-error>;
+}
+
+/// Interface for querying a Postgres database with prepared statements
+interface prepared {
+  use types.{pg-value, result-row, statement-prepare-error, prepared-statement-exec-error};
+
+  /// A token that represents a previously created prepared statement,
+  ///
+  /// This token can be expected to be somewhat opaque to users.
+  type prepared-statement-token = string;
+
+  /// Prepare a statement, given a connection token (which can represent a connection *or* session),
+  /// to a Postgres database.
+  ///
+  /// Queries *must* be parameterized, with named arguments in the form of `$<integer>`, for example:
+  ///
+  /// ```
+  /// SELECT email,username FROM users WHERE uuid=$1;
+  /// ```
+  ///
+  /// NOTE: To see how to obtain a `connection-token`, see `connection.wit`.
+  ///
+  prepare: func(
```

Comment:
as in the question about connection tokens, why is this represented by a token and not a resource?

Reply:
Hey @esoterra the reason here is the same -- this is a token because right now arbitrary resources aren't quite covered by our host yet! It requires a little extra integration which IMO is a large price to pay for achieving some notion of idiomatic WIT use!

That said -- likely before official `0.1.0` or *maybe* a `0.2.0` we'll address *both* async and resource usage, but the utility of the interface doesn't diminish by much in the meantime.